File: Backend/DealerAUTO.DTO/Models/Car.cs
using System.ComponentModel.DataAnnotations;

namespace DealerAUTO.DTO.Models;

public class Car : IModel<Guid>
{
    public Guid Id { get; set; }

    [Required]
    public string Make { get; set; } = null!;

    [Required]
    public string Model { get; set; } = null!;

    [Required]
    public int Year { get; set; }

    [Required]
    public int Mileage { get; set; }

    [Required]
    public string Description { get; set; } = null!;

    [Required]
    public string Fuel { get; set; } = null!;

    [Required]
    public int CubicCapacity { get; set; }

    [Required]
    public int Power { get; set; }

    [Required]
    public string Transmission { get; set; } = null!;

    [Required]
    public string Traction { get; set; } = null!;

    [Required]
    public string Body { get; set; } = null!;

    [Required]
    public string Wheel { get; set; } = null!;

    [Required]
    public int Price { get; set; }

    [Required]
    public string State { get; set; } = null!;

    [Required]
    public DateTime PostTime { get; set; }

    public Guid CarPost { get; set; }
    public ICollection<CarImage> Images { get; set; } = null!;
}

File: Frontend/src/app/Interfaces/Car.ts
import { SafeResourceUrl } from "@angular/platform-browser";

export interface Car {

  make: string;
  model: string;
  description: string;
  fuel: string;
  transmission: string;
  traction: string;
  body: string;
  wheel: string;
  state: string;
  year: number;
  mileage: number;
  cubicCapacity: number;
  power: number;
  price: number;
  postTime: Date;

  yearAsString: string;
  mileageAsString: string;
  cubicCapacityAsString: string;
  powerAsString: string;
  priceAsString: string;
  postTimeAsString: string;

  imagesAsBase64Strings: string[];
  firstImage: SafeResourceUrl;

}
